plutus-ledger/src/Ledger/Tx/Internal.hs

```diff
@@ -228,48 +229,52 @@ instance OpenApi.ToSchema TxOut where
 
 instance Pretty TxOut where
   pretty (TxOut (C.TxOut addr v d rs)) =
-    hang 2 $ vsep
+    hang 2 $ vsep $
       ["-" <+> pretty (fromCardanoTxOutValue v) <+> "addressed to"
       , pretty (fromCardanoAddressInEra addr)
-      , "with" <+> case fromCardanoTxOutDatum d of
-          PV2.NoOutputDatum      -> "no datum"
-          PV2.OutputDatum dv     -> "inline datum" <+> viaShow dv
-          PV2.OutputDatumHash dh -> "datum hash" <+> pretty dh
-      , "and with" <+> case rs of
-          C.ReferenceScript _ (C.ScriptInAnyLang _ s) ->
-            "reference script hash" <+> viaShow (C.hashScript s)
-          C.ReferenceScriptNone -> "no reference script"
       ]
+      <> case fromCardanoTxOutDatum d of
+          PV2.NoOutputDatum      -> []
+          PV2.OutputDatum dv     -> ["with inline datum" <+> viaShow dv]
+          PV2.OutputDatumHash dh -> ["with datum hash" <+> pretty dh]
+      <> case rs of
+          C.ReferenceScript _ (C.ScriptInAnyLang _ s) ->
+            ["with reference script hash" <+> viaShow (C.hashScript s)]
+          C.ReferenceScriptNone -> []
```

Comment:
Same here. "no reference script". What do you think?

Reply:
That overloads the logs quite a bit, I removed it because of that.